File: k9mail/src/test/java/com/fsck/k9/pEp/ui/keys/PepExtraKeysPresenterTest.java
package com.fsck.k9.pEp.ui.keys;

import com.fsck.k9.pEp.PEpProvider;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import java.util.Collections;
import java.util.Set;

import static org.mockito.Matchers.any;
import static org.mockito.Mockito.verify;

public class PepExtraKeysPresenterTest {

    private PepExtraKeysPresenter presenter;
    @Mock
    private PepExtraKeysView view;
    @Mock
    PEpProvider provider;

    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);
        presenter = new PepExtraKeysPresenter();
    }

    @Test
    public void shouldGetMasterKeysInfoWhenSetupMasterKeys() {
        presenter.initialize(view, provider, keys());

        verify(provider).getMasterKeysInfo();
    }

    @Test
    public void shouldShowKeysInfoWhenSetupMasterKeys() {
        presenter.initialize(view, provider, keys());

        verify(view).showKeys(any());
    }

    private Set<String> keys() {
        return Collections.emptySet();
    }
}

File: k9mail/src/test/java/com/fsck/k9/pEp/ui/keys/PepExtraKeysPresenterTest.kt
package com.fsck.k9.pEp.ui.keys

import com.fsck.k9.pEp.PEpProvider
import com.fsck.k9.pEp.testutils.CoroutineTestRule
import com.nhaarman.mockito_kotlin.any
import kotlinx.coroutines.ExperimentalCoroutinesApi
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.mockito.Mock
import org.mockito.Mockito
import org.mockito.MockitoAnnotations

@ExperimentalCoroutinesApi
class PepExtraKeysPresenterTest {
    @get:Rule
    var coroutinesTestRule = CoroutineTestRule()

    private lateinit var presenter: PepExtraKeysPresenter

    @Mock
    private lateinit var view: PepExtraKeysView

    @Mock
    private lateinit var provider: PEpProvider
    @Before
    @Throws(Exception::class)
    fun setUp() {
        MockitoAnnotations.openMocks(this)
        presenter = PepExtraKeysPresenter(coroutinesTestRule.testDispatcherProvider)
    }

    @Test
    @Throws(Exception::class)
    fun shouldGetMasterKeysInfoWhenSetupMasterKeys(): Unit =
            coroutinesTestRule.testDispatcher.run {
        presenter.initialize(view, provider, keys())
        Mockito.verify(provider).masterKeysInfo
    }

    @Test
    @Throws(Exception::class)
    fun shouldShowKeysInfoWhenSetupMasterKeys() =
            coroutinesTestRule.testDispatcher.run {
        presenter.initialize(view, provider, keys())
        Mockito.verify(view).showKeys(any())
    }

    private fun keys(): Set<String> {
        return emptySet()
    }
}